### PR TITLE
Refactor examples and enhance documentation in result.rs

### PR DESCRIPTION
**Pull request by renatoalvesca**
- Replaced `map` with `map_err` in the error handling example for correctness
- Reordered example code to improve readability and logical flow
- Added assertions to examples to demonstrate expected outcomes

### CODE REVIEW

File: library/core/src/result.rs [hunk 1]
```diff
@@ -45,25 +45,29 @@
...
 //! let good_result: Result<bool, i32> = good_result.and_then(|i| Ok(i == 11));
+//! assert_eq!(good_result, Ok(true));
 //!
 //! // Use `or_else` to handle the error.
 //! let bad_result: Result<i32, i32> = bad_result.or_else(|i| Ok(i + 20));
+//! assert_eq!(bad_result, Ok(29));
 //!
 //! // Consume the result and return the contents with `unwrap`.
 //! let final_awesome_result = good_result.unwrap();
+//! assert_eq!(final_awesome_result)
```
Comment (workingjubilee) [inline library/core/src/result.rs:70]: equal to?

Comment (prorealize) [reply library/core/src/result.rs]: Fixed

File: library/core/src/result.rs [hunk 2]
```diff
@@ -45,25 +45,29 @@
...
+//! // The `is_ok` and `is_err` methods do what they say.
 //! let good_result: Result<i32, i32> = Ok(10);
 //! let bad_result: Result<i32, i32> = Err(10);
-//!
-//! // The `is_ok` and `is_err` methods do what they say.
 //! assert!(good_result.is_ok() && !good_result.is_err());
 //! assert!(bad_result.is_err() && !bad_result.is_ok());
 //!
-//! // `map` consumes the `Result` and produces another.
+//! // `map` and `map_err` consume the `Result` and produces another.
```
Comment (joboet) [inline library/core/src/result.rs:54]:
```suggestion
//! // `map` and `map_err` consume the `Result` and produce another.
```

**Review by joboet — COMMENTED** [review]
Sure, why not! r=me[^1] after the typo is fixed.

[^1]: r=me is a shorthand we use to tell other reviewers that they can approve this on my behalf. You don't have to do anything about it.

**Comment by joboet** [comment]
Thanks!
@bors r+ rollup

**Pull request merged**